File: JavaScript/프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    obj = {};
    for (let idx = 0; idx < want.length; idx++){
        obj[want[idx]] = number[idx];
    }
    for (let cnt = 0; cnt <= discount.length - 10; cnt++){
        const tempObj = {...obj};
        for (let i = cnt; i < cnt + 10; i++){
            if (tempObj[discount[i]] !== undefined) tempObj[discount[i]]--;
        }
        if (Object.values(tempObj).every((v) => v === 0)) answer += 1;
    }
    return answer;
}

File: C++/프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

bool sum_mp(unordered_map<string, int> tmp_mp){
    for (auto mem: tmp_mp){
        if (mem.second != 0) return false; 
    }
    return true;
}

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    unordered_map<string, int> mp;
    
    for (int i = 0; i < number.size(); i++){
        mp[want[i]] = number[i];
    }// 다 넣어줬음.
    for (int i = 0; i < 10; i++){
        if (mp.count(discount[i])){
            mp[discount[i]] -= 1;
        }
    }
    if (sum_mp(mp)){
        answer += 1;
    }
    for (int i = 0; i < discount.size() - 10; i++){
        if (mp.count(discount[i])){
            mp[discount[i]] += 1;
        } 
        if (mp.count(discount[i + 10])){
            mp[discount[i + 10]] -= 1;
        }
        if (sum_mp(mp)){
            answer += 1;
        }
    }
    return answer;
}